### PR TITLE
docs: add theme toggle animation

### LINKED ISSUE #44644: 主题切换过渡效果
### What problem does this feature solve?

目前切换theme的token是没有任何动画效果的
希望可以增加动画效果

### What does the proposed API look like?
动画1
https://github.com/iyuli/Them-switch-animation-vue
参考这个demo，可以看到切换主题的动画效果

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close #44644

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Add theme toggle animation     |
| 🇨🇳 Chinese |    添加主题切换动画       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 49104ea</samp>

This pull request adds a circular clip-path animation for the theme switcher component in the documentation site. It uses a custom hook `useThemeAnimation` and the `startViewTransition` API to create a smooth and responsive transition between the light and dark themes. It also refactors the theme switcher component to use a boolean flag for the dark mode.

### <samp>🤖 Generated by Copilot at 49104ea</samp>

*  Add a custom hook `useThemeAnimation` to implement a circular clip-path animation for the theme switcher ([link](https://github.com/ant-design/ant-design/pull/44655/files?diff=unified&w=0#diff-941fcbacf1a38746130e95a5c46349e59999f00997963e17e027e79e2840a2c0R1-R91))
*  Use the hook in the `ThemeSwitch` component to trigger the animation on button click, passing the mouse event and the theme mode ([link](https://github.com/ant-design/ant-design/pull/44655/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447L25-R30), [link](https://github.com/ant-design/ant-design/pull/44655/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447L45-R55))
*  Simplify the theme mode check by using a variable `isDark` instead of `value.includes('dark')` ([link](https://github.com/ant-design/ant-design/pull/44655/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447L25-R30), [link](https://github.com/ant-design/ant-design/pull/44655/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447L45-R55))
*  Reorder some imports in the `ThemeSwitch` component for consistency ([link](https://github.com/ant-design/ant-design/pull/44655/files?diff=unified&w=0#diff-8d8f58ae995a6180f1788f819583cf51b719b8d8ab1ae422b06a1e8d846f2447L1-R9))

### CODE REVIEW

File: .dumi/hooks/useThemeAnimation.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const startAnimaTheme = () => {
+    const { isDark, event } = animateRef.current;
+    if (!event) return;
+    const x = event.clientX;
+    const y = event.clientY;
+    const endRadius = Math.hypot(Math.max(x, innerWidth - x), Math.max(y, innerHeight - y));
+
+    document
+      // @ts-ignore
+      .startViewTransition(() => {
```
Comment (yoyo837) [inline .dumi/hooks/useThemeAnimation.ts:50]: 不需要判断兼容吗?

Comment (RedJue) [reply .dumi/hooks/useThemeAnimation.ts:50]: 先布一版看看性能，兼容代码后面再补一下

Comment (yoyo837) [reply .dumi/hooks/useThemeAnimation.ts:50]: 懂了, 用预览镜像 😄

File: .dumi/hooks/useThemeAnimation.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  } = theme.useToken();
+  const animateRef = useRef<{
+    isDark: boolean;
+    event: MouseEvent | null;
+  }>({
+    isDark: false,
+    event: null,
+  });
+  // @ts-ignore
+  const isApiAvailable = useMemo(() => typeof document.startViewTransition === 'function', []);
```
Comment (kiner-tang) [inline .dumi/hooks/useThemeAnimation.ts:41]: 这里感觉没必要用 `useMemo`

File: .dumi/hooks/useThemeAnimation.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import { useEffect, useMemo, useRef } from 'react';
+
+import theme from '../../components/theme';
+
+const viewTransitionStyle = `
+    ::view-transition-old(root),
+    ::view-transition-new(root) {
+    animation: none;
+    mix-blend-mode: normal;
+    }
```
Comment (afc163) [inline .dumi/hooks/useThemeAnimation.ts:10]:
```suggestion
  ::view-transition-old(root),
  ::view-transition-new(root) {
    animation: none;
    mix-blend-mode: normal;
  }
```

缩进可以改改。

File: .dumi/hooks/useThemeAnimation.ts (hunk 2 above)
Comment (RedJue) [reply .dumi/hooks/useThemeAnimation.ts:41]: 可以去掉，顺手加了，确实没啥必要

File: .dumi/hooks/useThemeAnimation.ts (hunk 3 above)
Comment (RedJue) [reply .dumi/hooks/useThemeAnimation.ts:10]: 👌

File: .dumi/hooks/useThemeAnimation.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  };
+
+  const toggleAnimateTheme = (event: MouseEvent, isDark?: boolean) => {
+    animateRef.current.isDark = isDark;
+    animateRef.current.event = event;
+  };
+
+  // inject transition style
+  useEffect(() => {
+    if (isApiAvailable) {
```
Comment (zombieJ) [inline .dumi/hooks/useThemeAnimation.ts:82]: rc-util 有个 updateCSS 可以动态插入，不用手工创建 style

Comment (RedJue) [reply .dumi/hooks/useThemeAnimation.ts:82]: 👌 后面我修改一下

**Comment by MadCcc** [comment]
这个没必要 base feature 吧，文档更新直接 master 走起

**Review by zombieJ — APPROVED** [review]

File: .dumi/hooks/useThemeAnimation.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+          `circle(0px at ${x}px ${y}px)`,
+          `circle(${endRadius}px at ${x}px ${y}px)`,
+        ];
+        removeCSS('color-scheme');
+        startAnimationTheme(clipPath, isDark);
+      });
+  };
+
+  // inject transition style
+  useEffect(() => {
```
Comment (li-jia-nan) [inline .dumi/hooks/useThemeAnimation.ts:110]: 这个 `useEffect` 换成 `useInsertionEffect` 会不会更好一点？

**Comment by tuyangJs** [comment]
这个怎么使用，文档里没有写

**Pull request merged**